recipes/google-cloud-cpp-core/meta.yaml

```diff
@@ -0,0 +1,360 @@
+
+{%- macro child_devel_subpackage(feature, parent) %}
+script: install-libgoogle-cloud.sh   # [not win]
+script: install-libgoogle-cloud.bat  # [win]
+build:
+  run_exports:
+    # patch versions guaranteed to be ABI-compatible, see
+    # https://github.com/googleapis/google-cloud-cpp/blob/main/doc/adr/2023-05-03-patch-releases-are-ABI-compatible.md
+    - {{ pin_subpackage("libgoogle-cloud-" + feature, max_pin="x.x") }}
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+  host:
+    - {{ pin_subpackage("libgoogle-cloud-" + parent + "-devel", exact=True) }}
+    - {{ pin_subpackage("libgoogle-cloud-devel", exact=True) }}
+  run:
+    - {{ pin_subpackage("libgoogle-cloud-" + parent + "-devel", exact=True) }}
+    - {{ pin_subpackage("libgoogle-cloud-devel", exact=True) }}
+    - {{ pin_subpackage("libgoogle-cloud-" + feature, exact=True) }}
+    - libcxx  # [osx]
+test:
+  {{ test_devel_subpackage(feature)|indent(2) }}
+{%- endmacro %}
+
+package:
+  name: google-cloud-cpp-core
+  version: {{ version }}
+
+source:
+  url: https://github.com/googleapis/google-cloud-cpp/archive/v{{ version }}.tar.gz
+  sha256: 8cb87ec2947e691a7f8631877e78453bcfda51b3d3cd4940794a376741888d37
+  patches:
+    # This is an upstream fix to allow sharding google/cloud/oauth2
+    - patches/0001-fix-oauth2-only-enable-via-GOOGLE_CLOUD_CPP_ENABLE-1.patch
+    # This is an upstream fix to break the (unused) dependency because `storage`
+    # and ZLIB::ZLIB
+    - patches/0002-fix-storage-remove-unused-dependencies-12890.patch
+    # This is an upstream fix to use `find_package(CURL)` directly.
+    - patches/0003-cleanup-CMake-3.13-always-has-CURL-libcurl-12938.patch
+
+build:
+  number: 0
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+    - libgrpc
+    - libprotobuf
+  host:
+    - libabseil
+    - libcrc32c
+    - libcurl
+    - libgrpc
+    - libprotobuf
+    - openssl
+    - nlohmann_json
+    - zlib
+
+outputs:
+  - name: libgoogle-cloud
+    script: install-libgoogle-cloud.sh   # [not win]
+    script: install-libgoogle-cloud.bat  # [win]
+    build:
+      run_exports:
+        # patch versions guaranteed to be ABI-compatible, see
+        # https://github.com/googleapis/google-cloud-cpp/blob/main/doc/adr/2023-05-03-patch-releases-are-ABI-compatible.md
+        - {{ pin_subpackage("libgoogle-cloud", max_pin="x.x") }}
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - libgrpc
+        - libprotobuf
+      host:
+        - libabseil
+        - libcurl
+        - libgrpc
+        - libprotobuf
+        - openssl
+        - nlohmann_json
+        - libcxx  # [osx]
+      run:
+        - libcxx  # [osx]
+      run_constrained:
+        - libgoogle-cloud {{ version }} *_{{ PKG_BUILDNUM }}
+    test:
+      {{ test_subpackage("common")|indent(6) }}
```

Comment:
It works because one of the libraries installed by `libgoogle-cloud` matches all the requisites of a feature library.

That was too clever, I have expanded the macro here to make things easier to follow.